File: javascript/hamming/hamming.js
var compute = function (strand1, strand2) {
    if (strand1.length != strand2.length) {
        throw 'Strands must be the same length';
    }

    var intersect = strand1.split('').intersect(strand2.split(''));

    return strand1.length - intersect.length;
}

Array.prototype.intersect = function (compare) {
    var intersect = [];

    for (var i = 0; i < this.length; i++) {
        if (this[i] == compare[i]) {
            intersect.push(this[i]);
        }
    }

    return intersect;
}

module.exports = {
    compute : compute
};

File: go/hamming/hamming.go
package hamming

import "fmt"

const testVersion = 4

func Distance(a, b string) (int, error) {
	hamming := 0

	if len(a) != len(b) {
		return 0, fmt.Errorf("DNA strands must be same length")
	}

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			hamming++
		}
	}

	return hamming, nil
}
